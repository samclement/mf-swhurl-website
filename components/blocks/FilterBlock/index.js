import React from 'react'
import Link from '../../elements/Link'
import styled from 'styled-components'

const StyledBlock = styled.div`
  width: 150px;
  padding: 10px;
`

const StyledList = styled.ul`
  padding: 0;
  margin: 0;
`

const StyledLi = styled.li`
  list-style: none;
  padding-left: 0;
  margin-left: 0;
`

export default ({ title, filters }) => {
  return (
    <StyledBlock>
      <h2>{title}</h2>
      <StyledList>
        {filters.map((f, i) => {
          const url = `/plp?url=${f.url}`
          return (
            <StyledLi key={i}>
              <Link key={i} href={url} as={f.url} label={f.name} />
            </StyledLi>
          )
        })}
      </StyledList>
    </StyledBlock>
  )
}
