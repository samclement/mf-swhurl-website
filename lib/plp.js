module.exports.parseNavigation = navigation => {
  return navigation.map(item => {
    const { name, url } = item
    return { name, url }
  })
}

module.exports.parseSearchResults = searchResults => {
  return {
    categoryFacets: parseCategoryFacets(searchResults.categoryNavData),
    facets: parseFacets(searchResults.facets),
    results: parseResults(searchResults.results)
  }
}

function parseResults(results) {
  return results.map(parseProductResult)
}

function parseCategoryFacets(cFacets) {
  return cFacets.map(parseCategoryFacet)
}

function parseFacets(facets) {
  return facets
    .filter(facet => {
      return (
        facet.code === 'allDesigners' ||
        facet.code === 'refinementColour' ||
        facet.code === 'clothesSize'
      )
    })
    .map(parseFacet)
}

function parseCategoryFacet(cFacet) {
  const parsed = {
    name: cFacet.name,
    isSelected: cFacet.active,
    url: cFacet.url,
    categories: []
  }
  if (cFacet.active && cFacet.categories) {
    parsed.categories = cFacet.categories.map(parseCategoryFacet)
  }
  return parsed
}

function parseFacet(facet) {
  return {
    name: facet.name,
    values: facet.values.map(parseFacetValue)
  }
}

function parseFacetValue(facetValue) {
  return {
    name: facetValue.name,
    url: facetValue.query.url
  }
}

function parseProductResult(product) {
  return {
    name: product.name,
    url: product.url,
    code: product.code,
    price: product.price.value,
    imageCount: product.galleryImageMaps.length,
    sizes: product.sizeList.map(size => {
      return parseSize(size)
    })
  }
}

function parseSize(size) {
  return {
    code: size.code,
    baseCode: size.baseCode,
    description: size.value,
    inStock: size.available
  }
}
