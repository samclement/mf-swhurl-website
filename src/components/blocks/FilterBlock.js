import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

const StyledBlock = styled.div`
  width: 210px;
  padding: 10px;
`

const StyledFilters = styled.div`
  max-height: ${props => (props.scrollable ? '420px' : 'initial')};
  overflow-y: ${props => (props.scrollable ? 'auto' : 'initial')};
  border: ${props => (props.scrollable ? '1px solid #ccc' : 'initial')};
  padding: ${props => (props.scrollable ? '5px 9px' : 'initial')};
`

const StyledLink = styled.a`
  color: black;
  font-size: 13px;
  text-decoration: none;
  display: block;
  padding-left: 10px;
  text-indent: -15px;
  margin-bottom: 6px;
  ::before {
    content: ${props =>
      props.selected
        ? 'url("/checkbox_checked.svg")'
        : 'url("/checkbox_unchecked.svg")'};
    position: relative;
    bottom: -2px;
  }
`

export default ({ title, filters, scrollable }) => {
  return (
    <StyledBlock>
      <h2>{title}</h2>
      <StyledFilters scrollable={scrollable}>
        {filters.map((f, i) => {
          const url = `/plp?url=${f.url}`
          return (
            <Link key={i} href={url} as={f.url}>
              <StyledLink href={f.url} selected={f.selected}>
                {f.name}
              </StyledLink>
            </Link>
          )
        })}
      </StyledFilters>
    </StyledBlock>
  )
}
