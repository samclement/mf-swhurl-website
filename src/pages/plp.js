import Plp from '../components/containers/Plp'

export default Plp
