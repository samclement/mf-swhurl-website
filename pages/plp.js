import React, { useState, useEffect } from 'react'
import ProductListItem from '../components/blocks/ProductListItem'
import Nav from '../components/blocks/Nav'
import GenderHeader from '../components/elements/GenderHeader.js'
import FilterBlock from '../components/blocks/FilterBlock'
import Head from 'next/head'
import { mens, womens } from '../navigation.js'
import { getSearchResults } from '../services'
import styled from 'styled-components'
import CheeseburgerMenu from 'cheeseburger-menu'

const ProductsList = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
`

const SearchContainer = styled.div`
  display: flex;
  align-items: flex-start;
`

const FilterBlockContainer = styled.div`
  display: flex;
  flex-flow: column nowrap;
`

const DesktopFilterBlockContainer = styled(FilterBlockContainer)`
  @media only screen and (max-width: 640px) {
    display: none;
  }
`

const FilterButton = styled.button`
  @media only screen and (min-width: 640px) {
    display: none;
  }
`

function Plp(props) {
  const { gender, results, categoryFacets, facets } = props
  const [menuOpen, setMenuOpen] = useState(false)
  useEffect(() => {
    document.body.style.overflow = menuOpen ? 'hidden' : 'auto'
  })
  return (
    <div>
      <Head>
        <title>mf.swhurl.com | {gender}</title>
      </Head>
      <Nav items={mens} gender="mens" />
      <Nav items={womens} gender="womens" />
      <GenderHeader gender={gender} />
      <FilterButton onClick={() => setMenuOpen(!menuOpen)}>
        Show Filters
      </FilterButton>
      <SearchContainer>
        <CheeseburgerMenu
          isOpen={menuOpen}
          width={220}
          innerClassName="touch"
          closeCallback={() => setMenuOpen(false)}
        >
          <FilterBlockContainer>
            <FilterBlock title="Category" filters={categoryFacets} />
            {facets.map((f, i) => {
              return <FilterBlock key={i} title={f.name} filters={f.values} />
            })}
          </FilterBlockContainer>
        </CheeseburgerMenu>
        <DesktopFilterBlockContainer>
          <FilterBlock title="Category" filters={categoryFacets} />
          {facets.map((f, i) => {
            return <FilterBlock key={i} title={f.name} filters={f.values} />
          })}
        </DesktopFilterBlockContainer>
        <ProductsList>
          {results.map((item, i) => {
            return <ProductListItem key={i} item={item} />
          })}
        </ProductsList>
      </SearchContainer>
    </div>
  )
}

Plp.getInitialProps = async ({ req, query }) => {
  const q = req && req.query ? req.query : query
  const res = await getSearchResults(q, !!req)
  const { results, categoryFacets, facets } = res.data
  const gender = q.url.match(/\/mens/) ? 'mens' : 'womens'
  return { results, gender, categoryFacets, facets }
}

export default Plp
