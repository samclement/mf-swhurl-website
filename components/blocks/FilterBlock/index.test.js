/* eslint-env jest */

import React from 'react'
import { render } from 'enzyme'
import 'jest-styled-components'
import data from '../../../test-data/activewear.json'
import { parseSearchResults } from '../../../lib/plp.js'
import FilterBlock from './index.js'

const facets = parseSearchResults(data).facets
const facet = facets[0]

describe('With Snapshot Testing', () => {
  it('FilterBlock displays items', () => {
    const wrapper = render(<FilterBlock title={facet.name} filters={facet.values} />)
    expect(wrapper).toMatchSnapshot()
  })
})
