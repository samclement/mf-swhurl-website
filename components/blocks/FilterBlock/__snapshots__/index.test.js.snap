// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing FilterBlock displays items 1`] = `
.c3 {
  color: black;
  padding-right: 5px;
}

.c3:hover {
  color: #F00;
}

.c3:visited {
  color: black;
}

.c0 {
  width: 150px;
  padding: 10px;
}

.c1 {
  padding: 0;
  margin: 0;
}

.c2 {
  list-style: none;
  padding-left: 0;
  margin-left: 0;
}

<div
  class="c0"
>
  <h2>
    Designer
  </h2>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensalyx"
      >
        1017 ALYX 9SM
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amens-2xu"
      >
        2XU
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensandwander"
      >
        And Wander
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensaztech-mountain"
      >
        Aztech Mountain
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensblackbarrett-by"
      >
        Blackbarrett by Neil Barrett
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenscafe-du"
      >
        Caf√© Du Cycliste
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenscastore"
      >
        Castore
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensfalke"
      >
        Falke
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensfendi"
      >
        Fendi
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenshelly-hansen"
      >
        Helly Hansen
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensiffley-road"
      >
        Iffley Road
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenslndr"
      >
        LNDR
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmammut"
      >
        Mammut Delta X
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmoncler"
      >
        Moncler
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmoncler-genius"
      >
        Moncler Genius
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmoncler-grenoble"
      >
        Moncler Grenoble
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensnemen"
      >
        Nemen
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenspeak-performance"
      >
        Peak Performance
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensperfect-moment"
      >
        Perfect Moment
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenssalomon"
      >
        Salomon
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenssatisfy"
      >
        Satisfy
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensthe-upside"
      >
        The Upside
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenstilak"
      >
        Tilak
      </a>
    </li>
  </ul>
</div>
`;
